src/ui/posts/PostEditor.tsx

```diff
@@ -0,0 +1,91 @@
+import { ReactElement, useEffect, useState } from 'react';
+import { AppBus } from '@/bus/AppBus';
+import { Message } from '@/bus/Message';
+import { ContentBus } from '@/bus/ContentBus';
+import { Post, PostField } from '@/model/content/Post';
+import { FieldEditor } from '@/ui/posts/FieldEditor';
+
+interface Props {
+	post: Post;
+	fieldOrder: Record< string, number >;
+	onFieldChanged: ( name: string, field: PostField ) => void;
+}
+
+export function PostEditor( props: Props ) {
+	const { post, fieldOrder, onFieldChanged } = props;
+	const [ lastClickedElement, setLastClickedElement ] = useState< string >();
+	const [ fieldWaitingForSelection, setFieldWaitingForSelection ] = useState<
+		false | { field: PostField; name: string }
+	>( false );
+
+	// Listen to click events coming from the content script.
+	useEffect( () => {
+		AppBus.listen( async ( message: Message ) => {
+			switch ( message.action ) {
+				case AppBus.actions.ElementClicked:
+					await ContentBus.disableHighlighting();
+					setLastClickedElement( ( message.payload as any ).content );
+			}
+		} );
+		return () => {
+			void ContentBus.disableHighlighting();
+			AppBus.stopListening();
+		};
+	}, [] );
+
+	// Handle a click on an event in the content script,
+	// according to which field is currently waiting for selection.
+	useEffect(
+		() => {
+			if ( ! fieldWaitingForSelection || ! lastClickedElement ) {
+				return;
+			}
+			fieldWaitingForSelection.field.original = lastClickedElement;
+			onFieldChanged(
+				fieldWaitingForSelection.name,
+				fieldWaitingForSelection.field
+			);
+			setFieldWaitingForSelection( false );
+			setLastClickedElement( undefined );
```

Comment:
Should we perhaps define a `clearLastClickedElement()` than explicitly passing `undefined`?

Reply:
Maybe you're already aware of this, but in case you aren't: When you call `useState()`, react returns a variable that carries the value of that piece of state, and a function that you must used to update that state. So we have to use that function to modify the state. 

```ts
const [ lastClickedElement, setLastClickedElement ] = useState< string >();
```

I guess we could define a wrapper function that then calls `setLastClickedElement()`:

```ts
function clearLastClickedElement() {
    setLastClickedElement( undefined );
}
```

Is this what you meant?

I'm not sure this is a common pattern that is used, I don't remember seeing it. I think typically you call the `setFoo()` functions directly. The advantage of calling it directly is that it's immediately clear that state will be modified, while if we would be calling a wrapper function, that wouldn't be the case.

In any case, if you think we should add this wrapper function, I'm happy do so.

Reply:
I didn't realize that it was React's state method. My suggestion was based thinking of it as a general function. Ok to set "undefined" then 👍